internal/build/image_and_cache_builder.go

```diff
@@ -0,0 +1,175 @@
+package build
```

Comment:
fwiw, i probably would have left this in the same package...i feel like moving it to a different package forced you to make a bunch of odd architecture decisions around `mode`, which distracts from the main focus of this PR

Reply:
😅 legit!